modules/javafx.graphics/src/test/java/test/com/sun/javafx/css/StyleManagerTest.java

```diff
@@ -1109,4 +1109,15 @@ public void testConcurrentAccess() {
         assertFalse("Exception during CSS processing on BG thread", err.get());
     }
 
+    @Test
+    public void testCalculateCheckSum() {
+        StyleManagerShim sm = StyleManagerShim.getInstance();
+        String expectedChecksum = "03eadaed6c9269d273c3ffa670d14ff8"; // Generated using command 'md5sum checksum.css'
+        String actualChecksum = "";
+        byte[] checksum = sm.calculateCheckSum(getClass().getResource("checksum.css").toExternalForm());
+        for (byte b : checksum) {
+            actualChecksum += Integer.toString((b & 0xff) + 0x100, 16).substring(1);
+        }
+        assertEquals(expectedChecksum, actualChecksum);
```

Comment:
It might be clearer to use a `byte` array for the expected checksum and use `Arrays.equals()` to compare them rather than converting to a `String`. I'm scratching my head over the formula in the `Integer.toString` call.

Reply:
I have updated the test to use a byte array. Some byte values are greater than 128, those are the negative values.